vendor/bat-native-ads/src/bat/ads/internal/ads/serving/choose/eligible_ads_predictor_util.h

```diff
@@ -159,6 +161,35 @@ CreativeAdPredictorMap<T> ComputePredictorFeaturesAndScores(
   return creative_ad_predictors_with_features;
 }
 
+template <typename T>
+std::vector<int> ComputeVoteRegistry(
+    const std::vector<T>& creative_ads,
+    const TextEmbeddingHtmlEventList& text_embedding_html_events) {
+  std::vector<int> vote_registry;
+  vote_registry.assign(creative_ads.size(), 0);
+
+  for (const auto& text_embedding : text_embedding_html_events) {
+    int max_index = 0;
+    float max_similarity = 0.0;
+    for (const auto& creative_ad : creative_ads) {
+      ml::VectorData ad_embedding(creative_ad.embedding);
+      const ml::VectorData page_text_embedding(text_embedding.embedding);
+      const float similarity_score =
+          ad_embedding.ComputeSimilarity(page_text_embedding);
+
+      if (similarity_score > max_similarity) {
+        max_index =
+            std::find(creative_ads.cbegin(), creative_ads.cend(), creative_ad) -
+            creative_ads.begin();
+        max_similarity = similarity_score;
+      }
+    }
+    vote_registry[max_index]++;
```

Comment:
max_index defaults to 0. So if the size of creative_ads is 0, it would still tally a `vote_registry` vote for index 0

Reply:
as discussed in slack, `ComputeVoteRegistry` should not be passed in an empty creative ads vector, as there would be nothing to compute votes on. So I added an early exit before `ComputeVoteRegistry` gets called in `predict_ad_embeddings` and a DCHECK on the `creative_ads` argument within the function